src/controller/AutoCommissioner.cpp

```diff
@@ -496,9 +496,24 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStageInternal(Commissio
         }
         return CommissioningStage::kFindOperational;
     case CommissioningStage::kFindOperational:
-        return CommissioningStage::kICDSendStayActive;
+        if (mActionOverCase == ActionOverCase::kICDSendStayActive)
```

Comment:
This would be cleaner as just two separate states: kFindOperationalForStayActive and kFindOperationalForCommissioningComplete, instead of having the same state but then also some out-of-band state.

Reply:
updated, thanks